lib/Backend/Vault/Kv/Internal.hs

```diff
@@ -0,0 +1,311 @@
+import           Servant.API
+import           Servant.Client
+import           Servant.API.Generic
+
+-- |
+-- A type defining most responses from the Vault server, it wraps the data we're
+-- actually trying to read as some sort of super class.
+-- For example:
+--
+-- >  { "request_id": "00000000-0000-0000-0000-000000000000"
+-- >  , "lease_id": ""
+-- >  , "renewable": false
+-- >  , "lease_duration": 0
+-- >  , "data": <DATA>
+-- >  , "wrap_info": null
+-- >  , "warnings": null
+-- >  , "auth": null
+-- >  }
+data KvResponse a =
+  KvResponse
+  { _requestId :: T.Text
+  , _leaseId :: T.Text
+  , _renewable :: Bool
+  , _leaseDuration :: Int
+  , _kdata :: a
+  }
+  deriving (Show)
+makeLenses ''KvResponse
+
+
+-- |
+-- A type defining the response from the Vault server at endpoint
+-- '/v1/<MOUNT>/metadata/<PATH>'. This is all actually always wrapped
+-- in 'KvResponse' at the 'data' key.
+-- For example:
+--
+-- >  { "keys":
+-- >    [ "a1"
+-- >    , "q1"
+-- >    ]
+-- >  }
+newtype ListSecrets = ListSecrets { _unListSecrets :: [T.Text] }
+  deriving (Show)
+makeLenses ''ListSecrets
+
+-- |
+-- A type defining the response from the Vault server at endpoint
+-- '/v1/<MOUNT>/data/<PATH>?version=<VERSION>'. This is all actually always wrapped
+-- in 'KvResponse' at the 'data' key.
+-- For example:
+--
+-- >  { "data":
+-- >    { "token": "stuff"
+-- >    }
+-- >  , "metadata": <ReadSecretMetadata>
+-- >    { "created_time": "2018-03-22T02:24:06.945319214Z",
+-- >      "custom_metadata":
+-- >      { "owner": "jdoe",
+-- >        "mission_critical": "false"
+-- >      },
+-- >      "deletion_time": "",
+-- >      "destroyed": false,
+-- >      "version": 2
+-- >    }
+-- >  }
+data ReadSecret =
+  ReadSecret
+  { _rsSecret :: HS.HashMap T.Text T.Text
+  , _rsCustomMetadata :: HS.HashMap T.Text T.Text
+  , _rsCreatedTime :: T.Text
+  , _rsDeletionTime :: T.Text
+  , _rsDestroyed :: Bool
+  , _rsVersion :: Int
+  }
+  deriving (Show)
+makeLensesWith abbreviatedFields ''ReadSecret
+
+
+-- |
+-- A type defining the request to the Vault servet at endpoint
+-- '/v1/<MOUNT>/data/<PATH>'.
+-- For example:
+--
+-- >  { "options":
+-- >  , { "cas": 0
+-- >    }
+-- >  , "data":
+-- >    { "foo": "bar"
+-- >    , "zip": "zap"
+-- >    }
+-- >  }
+data PostSecret =
+  PostSecret
+  { _psCas :: Maybe Int
+  , _psDdata :: HS.HashMap T.Text T.Text
+  }
+  deriving (Show)
+type PatchSecret = PostSecret
+makeLensesWith abbreviatedFields ''PostSecret
+
+
+-- |
+-- A type defining the request to the Vault servet at endpoint
+-- '/v1/<MOUNT>/metadata/<PATH>'.
+-- For example:
+--
+-- >  { "max_versions": 5
+-- >  , "cas_required": false
+-- >  , "delete_version_after": "40m"
+-- >  , "custom_metadata":
+-- >    { "example": ":)"
+-- >    }
+-- >  }
+data UpdateMetadata =
+  UpdateMetadata
+  { _umMaxVersions :: Maybe Int
+  , _umCasRequired :: Maybe Bool
+  , _umDeleteVersionAfter :: Maybe T.Text
+  , _umCustomMetadata :: HS.HashMap T.Text T.Text
+  }
```

Comment:
There's no need for the `_` prefix anymore.

When using `makeLenses`, it makes sense to add a `_` prefix.
`makeLenses` will detect a field named `_x` and create a lens named `x`. See the example in the [docs here](https://hackage.haskell.org/package/lens-5.1/docs/Control-Lens-Combinators.html#v:makeLenses),

But because we're now using `makeLensesWith abbreviatedFields` with an `um` prefix, there's no need for an _additional_  `_` prefix.

Reply:
Done

Reply:
Please do the same (use an abbreviated prefix + switch to `makeLensesWith`) for `FieldMetadata`, `CofferSpecials`, `KvResponse` and `Routes`.

Reply:
Oh and remove the `_` prefix from `VaultKvBackend` please.

Reply:
Done